Makefile

```
@@ -53,7 +53,7 @@ tests: backend_tests
 backend_tests:
 	@echo "🍜 Running python tests"
 	${DOCKER_COMPOSE_TEST} up -d neo4j
-	${DOCKER_COMPOSE_TEST}  run --rm taxonomy_api pytest . /parser
+	${DOCKER_COMPOSE_TEST}  run --rm taxonomy_api pytest . //parser
```

this is incompatible with linux and Mac OS run.

Why not:
```suggestion
	${DOCKER_COMPOSE_TEST}  run --rm taxonomy_api pytest . "./parser"
```
Doesn't this work on windows ?

---

Hey @alexgarel 

The syntax you have suggested does not work. It throws the directory not found error.

```
COMPOSE_PROJECT_NAME=test_taxonomy docker-compose --env-file=.env  run --rm taxonomy_api pytest . "./parser"
ERROR: file or directory not found: ./parser
```

---

Ok, I tested and the `//parser` works on linux, so we can commit it.

---

But it breaks the CI :-(

---

@Jagrutiti can you try with `\/parser` ? (both on windows and if ok, commit so that we see if it's ok in CI)

---

Hey @alexgarel 

Unfortunately, `\/parser` does not work in Windows :(

Sharing the logs:

```
COMPOSE_PROJECT_NAME=test_taxonomy docker-compose --env-file=.env  run --rm taxonomy_api pytest . \/parser
================================================================== test session starts ===================================================================
platform linux -- Python 3.9.16, pytest-7.1.2, pluggy-1.0.0
rootdir: /code
plugins: anyio-3.6.1
collected 0 items

================================================================= no tests ran in 0.03s ==================================================================
ERROR: file or directory not found: C:/Program Files/Git/parser

make: *** [backend_tests] Error 4
```

---

@Jagrutiti another possible fix:

can you leave the Makefile as it was before, but add, at the beginning (where we define different variables): `MSYS_NO_PATHCONV=1` 
(taken from https://stackoverflow.com/a/34386471/2886726)

---

Hey @alexgarel 

The good news is this works:

```
backend_tests:
	@echo "🍜 Running python tests"
	${DOCKER_COMPOSE_TEST} up -d neo4j
	MSYS_NO_PATHCONV=1 ${DOCKER_COMPOSE_TEST}  run --rm taxonomy_api pytest . /parser
	${DOCKER_COMPOSE_TEST} stop neo4j
```

Sharing the complete logs for future reference:

```
$ make checks
ðŸœ Quality checks python
docker-compose --env-file=.env run --rm taxonomy_api flake8 .
docker-compose --env-file=.env run --rm taxonomy_api isort --check-only .
docker-compose --env-file=.env run --rm taxonomy_api black --check .
All done! ✨ 🍰 ✨
8 files would be left unchanged.
ðŸœ Quality checks JS
docker-compose --env-file=.env run --rm taxonomy_node npx prettier -c src/
Checking formatting...
All matched files use Prettier code style!
docker-compose --env-file=.env run --rm -e CI=true taxonomy_node npm run build

> taxonomy-editor-frontend@0.1.0 build
> react-scripts build

Creating an optimized production build...
Browserslist: caniuse-lite is outdated. Please run:
  npx update-browserslist-db@latest
  Why you should do it regularly: https://github.com/browserslist/update-db#readme
Compiled with warnings.

Failed to parse source map from '/opt/taxonomy-editor/node_modules/react-double-scrollbar/dist/DoubleScrollbar.js.map' file: Error: ENOENT: no such file or directory, open '/opt/taxonomy-editor/node_modules/react-double-scrollbar/dist/DoubleScrollbar.js.map'

Search for the keywords to learn more about each warning.
To ignore, add // eslint-disable-next-line to the line before.
  serve -s build

Find out more about deployment here:

  https://cra.link/deployment

git checkout taxonomy-editor-frontend/build/.empty || true
Updated 1 path from the index
ðŸœ Running python tests
COMPOSE_PROJECT_NAME=test_taxonomy NEO4J_ADMIN_EXPOSE=127.0.0.1:7475 NEO4J_BOLT_EXPOSE=127.0.0.1:7688 docker-compose --env-file=.env up -d neo4j   
[+] Running 1/1
 - Container test_taxonomy-neo4j-1  Started                                                                                                   1.5s 
MSYS_NO_PATHCONV=1 COMPOSE_PROJECT_NAME=test_taxonomy NEO4J_ADMIN_EXPOSE=127.0.0.1:7475 NEO4J_BOLT_EXPOSE=127.0.0.1:7688 docker-compose --env-file=.env  run --rm taxonomy_api pytest . /parser
=============================================================== test session starts ===============================================================platform linux -- Python 3.9.16, pytest-7.1.2, pluggy-1.0.0
rootdir: /
plugins: anyio-3.6.1
collected 7 items                                                                                                                                  

../parser/tests/integration/test_parse_unparse_integration.py EE                                                                            [ 28%] 
../parser/tests/integration/test_parser_integration.py EE                                                                                   [ 57%] 
../parser/tests/unit/test_parser_unit.py EE.                                                                                                [100%]

===================================================================== ERRORS ======================================================================
________________________________________________________ ERROR at setup of test_round_trip ________________________________________________________

address = IPv4Address(('neo4j', 7687)), family = 0

    @staticmethod
    def _dns_resolver(address, family=0):
        """ Regular DNS resolver. Takes an address object and optional
        address family for filtering.
    
        :param address:
        :param family:
        :returns:
        """
        try:
>           info = NetworkUtil.get_address_info(
                address.host, address.port, family=family,
                type=socket.SOCK_STREAM
            )

/usr/local/lib/python3.9/site-packages/neo4j/_async_compat/network/_util.py:114:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

host = 'neo4j', port = 7687

    @staticmethod
    def get_address_info(host, port, *, family=0, type=0, proto=0, flags=0):
>       return socket.getaddrinfo(host, port, family, type, proto, flags)

/usr/local/lib/python3.9/site-packages/neo4j/_async_compat/network/_util.py:102:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

host = 'neo4j', port = 7687, family = 0, type = <SocketKind.SOCK_STREAM: 1>, proto = 0, flags = 0

    def getaddrinfo(host, port, family=0, type=0, proto=0, flags=0):
        """Resolve host and port into list of address info entries.

        Translate the host/port argument into a sequence of 5-tuples that contain
        all the necessary arguments for creating a socket connected to that service.
        host is a domain name, a string representation of an IPv4/v6 address or
        None. port is a string service name such as 'http', a numeric port number or
        None. By passing None as the value of host and port, you can pass NULL to
        the underlying C API.

        The family, type and proto arguments can be optionally specified in order to
        narrow the list of addresses returned. Passing zero as a value for each of
        these arguments selects the full range of results.
        """
        # We override this function since we want to translate the numeric family
        # and socket type values to enum constants.
        addrlist = []
>       for res in _socket.getaddrinfo(host, port, family, type, proto, flags):
E       socket.gaierror: [Errno -3] Temporary failure in name resolution

/usr/local/lib/python3.9/socket.py:954: gaierror

During handling of the above exception, another exception occurred:

    @pytest.fixture
    def neo4j():
        """waiting for neo4j to be ready"""
        uri = os.environ.get("NEO4J_URI", "bolt://localhost:7687")
        driver = GraphDatabase.driver(uri)
        session = driver.session()
        connected = False
        while not connected:
            try:
>               session.run("MATCH () return 1 limit 1")

/parser/tests/conftest.py:18:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
/usr/local/lib/python3.9/site-packages/neo4j/_sync/work/session.py:280: in run
    self._connect(self._config.default_access_mode)
/usr/local/lib/python3.9/site-packages/neo4j/_sync/work/session.py:120: in _connect
    super()._connect(access_mode, **access_kwargs)
/usr/local/lib/python3.9/site-packages/neo4j/_sync/work/workspace.py:168: in _connect
    self._connection = self._pool.acquire(**acquire_kwargs_)
/usr/local/lib/python3.9/site-packages/neo4j/_sync/io/_pool.py:411: in acquire
    return self._acquire(
/usr/local/lib/python3.9/site-packages/neo4j/_sync/io/_pool.py:221: in _acquire
    return connection_creator()
/usr/local/lib/python3.9/site-packages/neo4j/_sync/io/_pool.py:138: in connection_creator
    connection = self.opener(
/usr/local/lib/python3.9/site-packages/neo4j/_sync/io/_pool.py:386: in opener
    return Bolt.open(
/usr/local/lib/python3.9/site-packages/neo4j/_sync/io/_bolt.py:322: in open
    BoltSocket.connect(
/usr/local/lib/python3.9/site-packages/neo4j/_async_compat/network/_bolt_socket.py:649: in connect
    for resolved_address in resolved_addresses:
/usr/local/lib/python3.9/site-packages/neo4j/_async_compat/network/_util.py:157: in resolve_address
    for resolved_address in NetworkUtil._dns_resolver(
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

address = IPv4Address(('neo4j', 7687)), family = 0

    @staticmethod
    def _dns_resolver(address, family=0):
        """ Regular DNS resolver. Takes an address object and optional
        address family for filtering.

        :param address:
        :param family:
        :returns:
        """
        try:
            info = NetworkUtil.get_address_info(
                address.host, address.port, family=family,
                type=socket.SOCK_STREAM
            )
        except OSError:
>           raise ValueError("Cannot resolve address {}".format(address))
E           ValueError: Cannot resolve address neo4j:7687

/usr/local/lib/python3.9/site-packages/neo4j/_async_compat/network/_util.py:119: ValueError
__________________________________________________ ERROR at setup of test_two_branch_round_trip ___________________________________________________

address = IPv4Address(('neo4j', 7687)), family = 0

    @staticmethod
    def _dns_resolver(address, family=0):
        """ Regular DNS resolver. Takes an address object and optional
        address family for filtering.

        :param address:
        :param family:
        :returns:
        """
        try:
>           info = NetworkUtil.get_address_info(
                address.host, address.port, family=family,
                type=socket.SOCK_STREAM
            )

/usr/local/lib/python3.9/site-packages/neo4j/_async_compat/network/_util.py:114:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

host = 'neo4j', port = 7687

    @staticmethod
    def get_address_info(host, port, *, family=0, type=0, proto=0, flags=0):
>       return socket.getaddrinfo(host, port, family, type, proto, flags)

/usr/local/lib/python3.9/site-packages/neo4j/_async_compat/network/_util.py:102:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

host = 'neo4j', port = 7687, family = 0, type = <SocketKind.SOCK_STREAM: 1>, proto = 0, flags = 0

    def getaddrinfo(host, port, family=0, type=0, proto=0, flags=0):
        """Resolve host and port into list of address info entries.

        Translate the host/port argument into a sequence of 5-tuples that contain
        all the necessary arguments for creating a socket connected to that service.
        host is a domain name, a string representation of an IPv4/v6 address or
        None. port is a string service name such as 'http', a numeric port number or
        None. By passing None as the value of host and port, you can pass NULL to
        the underlying C API.

        The family, type and proto arguments can be optionally specified in order to
        narrow the list of addresses returned. Passing zero as a value for each of
        these arguments selects the full range of results.
        """
        # We override this function since we want to translate the numeric family
        # and socket type values to enum constants.
        addrlist = []
>       for res in _socket.getaddrinfo(host, port, family, type, proto, flags):
E       socket.gaierror: [Errno -3] Temporary failure in name resolution

/usr/local/lib/python3.9/socket.py:954: gaierror

During handling of the above exception, another exception occurred:

    @pytest.fixture
    def neo4j():
        """waiting for neo4j to be ready"""
        uri = os.environ.get("NEO4J_URI", "bolt://localhost:7687")
        driver = GraphDatabase.driver(uri)
        session = driver.session()
        connected = False
        while not connected:
            try:
>               session.run("MATCH () return 1 limit 1")

/parser/tests/conftest.py:18:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
/usr/local/lib/python3.9/site-packages/neo4j/_sync/work/session.py:280: in run
    self._connect(self._config.default_access_mode)
/usr/local/lib/python3.9/site-packages/neo4j/_sync/work/session.py:120: in _connect
    super()._connect(access_mode, **access_kwargs)
/usr/local/lib/python3.9/site-packages/neo4j/_sync/work/workspace.py:168: in _connect
    self._connection = self._pool.acquire(**acquire_kwargs_)
/usr/local/lib/python3.9/site-packages/neo4j/_sync/io/_pool.py:411: in acquire
    return self._acquire(
/usr/local/lib/python3.9/site-packages/neo4j/_sync/io/_pool.py:221: in _acquire
    return connection_creator()
/usr/local/lib/python3.9/site-packages/neo4j/_sync/io/_pool.py:138: in connection_creator
    connection = self.opener(
/usr/local/lib/python3.9/site-packages/neo4j/_sync/io/_pool.py:386: in opener
    return Bolt.open(
/usr/local/lib/python3.9/site-packages/neo4j/_sync/io/_bolt.py:322: in open
    BoltSocket.connect(
/usr/local/lib/python3.9/site-packages/neo4j/_async_compat/network/_bolt_socket.py:649: in connect
    for resolved_address in resolved_addresses:
/usr/local/lib/python3.9/site-packages/neo4j/_async_compat/network/_util.py:157: in resolve_address
    for resolved_address in NetworkUtil._dns_resolver(
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

address = IPv4Address(('neo4j', 7687)), family = 0

    @staticmethod
    def _dns_resolver(address, family=0):
        """ Regular DNS resolver. Takes an address object and optional
        address family for filtering.

        :param address:
        :param family:
        :returns:
        """
        try:
            info = NetworkUtil.get_address_info(
                address.host, address.port, family=family,
                type=socket.SOCK_STREAM
            )
        except OSError:
>           raise ValueError("Cannot resolve address {}".format(address))
E           ValueError: Cannot resolve address neo4j:7687

/usr/local/lib/python3.9/site-packages/neo4j/_async_compat/network/_util.py:119: ValueError
_________________________________________________________ ERROR at setup of test_calling __________________________________________________________

address = IPv4Address(('neo4j', 7687)), family = 0

    @staticmethod
    def _dns_resolver(address, family=0):
        """ Regular DNS resolver. Takes an address object and optional
        address family for filtering.

        :param address:
        :param family:
        :returns:
        """
        try:
>           info = NetworkUtil.get_address_info(
                address.host, address.port, family=family,
                type=socket.SOCK_STREAM
            )

/usr/local/lib/python3.9/site-packages/neo4j/_async_compat/network/_util.py:114:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

host = 'neo4j', port = 7687

    @staticmethod
    def get_address_info(host, port, *, family=0, type=0, proto=0, flags=0):
>       return socket.getaddrinfo(host, port, family, type, proto, flags)

/usr/local/lib/python3.9/site-packages/neo4j/_async_compat/network/_util.py:102:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

host = 'neo4j', port = 7687, family = 0, type = <SocketKind.SOCK_STREAM: 1>, proto = 0, flags = 0

    def getaddrinfo(host, port, family=0, type=0, proto=0, flags=0):
        """Resolve host and port into list of address info entries.

        Translate the host/port argument into a sequence of 5-tuples that contain
        all the necessary arguments for creating a socket connected to that service.
        host is a domain name, a string representation of an IPv4/v6 address or
        None. port is a string service name such as 'http', a numeric port number or
        None. By passing None as the value of host and port, you can pass NULL to
        the underlying C API.

        The family, type and proto arguments can be optionally specified in order to
        narrow the list of addresses returned. Passing zero as a value for each of
        these arguments selects the full range of results.
        """
        # We override this function since we want to translate the numeric family
        # and socket type values to enum constants.
        addrlist = []
>       for res in _socket.getaddrinfo(host, port, family, type, proto, flags):
E       socket.gaierror: [Errno -3] Temporary failure in name resolution

/usr/local/lib/python3.9/socket.py:954: gaierror

During handling of the above exception, another exception occurred:

    @pytest.fixture
    def neo4j():
        """waiting for neo4j to be ready"""
        uri = os.environ.get("NEO4J_URI", "bolt://localhost:7687")
        driver = GraphDatabase.driver(uri)
        session = driver.session()
        connected = False
        while not connected:
            try:
>               session.run("MATCH () return 1 limit 1")

/parser/tests/conftest.py:18:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
/usr/local/lib/python3.9/site-packages/neo4j/_sync/work/session.py:280: in run
    self._connect(self._config.default_access_mode)
/usr/local/lib/python3.9/site-packages/neo4j/_sync/work/session.py:120: in _connect
    super()._connect(access_mode, **access_kwargs)
/usr/local/lib/python3.9/site-packages/neo4j/_sync/work/workspace.py:168: in _connect
    self._connection = self._pool.acquire(**acquire_kwargs_)
/usr/local/lib/python3.9/site-packages/neo4j/_sync/io/_pool.py:411: in acquire
    return self._acquire(
/usr/local/lib/python3.9/site-packages/neo4j/_sync/io/_pool.py:221: in _acquire
    return connection_creator()
/usr/local/lib/python3.9/site-packages/neo4j/_sync/io/_pool.py:138: in connection_creator
    connection = self.opener(
/usr/local/lib/python3.9/site-packages/neo4j/_sync/io/_pool.py:386: in opener
    return Bolt.open(
/usr/local/lib/python3.9/site-packages/neo4j/_sync/io/_bolt.py:322: in open
    BoltSocket.connect(
/usr/local/lib/python3.9/site-packages/neo4j/_async_compat/network/_bolt_socket.py:649: in connect
    for resolved_address in resolved_addresses:
/usr/local/lib/python3.9/site-packages/neo4j/_async_compat/network/_util.py:157: in resolve_address
    for resolved_address in NetworkUtil._dns_resolver(
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

address = IPv4Address(('neo4j', 7687)), family = 0

    @staticmethod
    def _dns_resolver(address, family=0):
        """ Regular DNS resolver. Takes an address object and optional
        address family for filtering.

        :param address:
        :param family:
        :returns:
        """
        try:
            info = NetworkUtil.get_address_info(
                address.host, address.port, family=family,
                type=socket.SOCK_STREAM
            )
        except OSError:
>           raise ValueError("Cannot resolve address {}".format(address))
E           ValueError: Cannot resolve address neo4j:7687

/usr/local/lib/python3.9/site-packages/neo4j/_async_compat/network/_util.py:119: ValueError
________________________________________________________ ERROR at setup of test_error_log _________________________________________________________

address = IPv4Address(('neo4j', 7687)), family = 0

    @staticmethod
    def _dns_resolver(address, family=0):
        """ Regular DNS resolver. Takes an address object and optional
        address family for filtering.

        :param address:
        :param family:
        :returns:
        """
        try:
>           info = NetworkUtil.get_address_info(
                address.host, address.port, family=family,
                type=socket.SOCK_STREAM
            )

/usr/local/lib/python3.9/site-packages/neo4j/_async_compat/network/_util.py:114:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

host = 'neo4j', port = 7687

    @staticmethod
    def get_address_info(host, port, *, family=0, type=0, proto=0, flags=0):
>       return socket.getaddrinfo(host, port, family, type, proto, flags)

/usr/local/lib/python3.9/site-packages/neo4j/_async_compat/network/_util.py:102:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

host = 'neo4j', port = 7687, family = 0, type = <SocketKind.SOCK_STREAM: 1>, proto = 0, flags = 0

    def getaddrinfo(host, port, family=0, type=0, proto=0, flags=0):
        """Resolve host and port into list of address info entries.

        Translate the host/port argument into a sequence of 5-tuples that contain
        all the necessary arguments for creating a socket connected to that service.
        host is a domain name, a string representation of an IPv4/v6 address or
        None. port is a string service name such as 'http', a numeric port number or
        None. By passing None as the value of host and port, you can pass NULL to
        the underlying C API.

        The family, type and proto arguments can be optionally specified in order to
        narrow the list of addresses returned. Passing zero as a value for each of
        these arguments selects the full range of results.
        """
        # We override this function since we want to translate the numeric family
        # and socket type values to enum constants.
        addrlist = []
>       for res in _socket.getaddrinfo(host, port, family, type, proto, flags):
E       socket.gaierror: [Errno -3] Temporary failure in name resolution

/usr/local/lib/python3.9/socket.py:954: gaierror

During handling of the above exception, another exception occurred:

    @pytest.fixture
    def neo4j():
        """waiting for neo4j to be ready"""
        uri = os.environ.get("NEO4J_URI", "bolt://localhost:7687")
        driver = GraphDatabase.driver(uri)
        session = driver.session()
        connected = False
        while not connected:
            try:
>               session.run("MATCH () return 1 limit 1")

/parser/tests/conftest.py:18:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
/usr/local/lib/python3.9/site-packages/neo4j/_sync/work/session.py:280: in run
    self._connect(self._config.default_access_mode)
/usr/local/lib/python3.9/site-packages/neo4j/_sync/work/session.py:120: in _connect
    super()._connect(access_mode, **access_kwargs)
/usr/local/lib/python3.9/site-packages/neo4j/_sync/work/workspace.py:168: in _connect
    self._connection = self._pool.acquire(**acquire_kwargs_)
/usr/local/lib/python3.9/site-packages/neo4j/_sync/io/_pool.py:411: in acquire
    return self._acquire(
/usr/local/lib/python3.9/site-packages/neo4j/_sync/io/_pool.py:221: in _acquire
    return connection_creator()
/usr/local/lib/python3.9/site-packages/neo4j/_sync/io/_pool.py:138: in connection_creator
    connection = self.opener(
/usr/local/lib/python3.9/site-packages/neo4j/_sync/io/_pool.py:386: in opener
    return Bolt.open(
/usr/local/lib/python3.9/site-packages/neo4j/_sync/io/_bolt.py:322: in open
    BoltSocket.connect(
/usr/local/lib/python3.9/site-packages/neo4j/_async_compat/network/_bolt_socket.py:649: in connect
    for resolved_address in resolved_addresses:
/usr/local/lib/python3.9/site-packages/neo4j/_async_compat/network/_util.py:157: in resolve_address
    for resolved_address in NetworkUtil._dns_resolver(
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

address = IPv4Address(('neo4j', 7687)), family = 0

    @staticmethod
    def _dns_resolver(address, family=0):
        """ Regular DNS resolver. Takes an address object and optional
        address family for filtering.

        :param address:
        :param family:
        :returns:
        """
        try:
            info = NetworkUtil.get_address_info(
                address.host, address.port, family=family,
                type=socket.SOCK_STREAM
            )
        except OSError:
>           raise ValueError("Cannot resolve address {}".format(address))
E           ValueError: Cannot resolve address neo4j:7687

/usr/local/lib/python3.9/site-packages/neo4j/_async_compat/network/_util.py:119: ValueError
___________________________________________________ ERROR at setup of test_normalized_filename ____________________________________________________

address = IPv4Address(('neo4j', 7687)), family = 0

    @staticmethod
    def _dns_resolver(address, family=0):
        """ Regular DNS resolver. Takes an address object and optional
        address family for filtering.

        :param address:
        :param family:
        :returns:
        """
        try:
>           info = NetworkUtil.get_address_info(
                address.host, address.port, family=family,
                type=socket.SOCK_STREAM
            )

/usr/local/lib/python3.9/site-packages/neo4j/_async_compat/network/_util.py:114:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

host = 'neo4j', port = 7687

    @staticmethod
    def get_address_info(host, port, *, family=0, type=0, proto=0, flags=0):
>       return socket.getaddrinfo(host, port, family, type, proto, flags)

/usr/local/lib/python3.9/site-packages/neo4j/_async_compat/network/_util.py:102:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

host = 'neo4j', port = 7687, family = 0, type = <SocketKind.SOCK_STREAM: 1>, proto = 0, flags = 0

    def getaddrinfo(host, port, family=0, type=0, proto=0, flags=0):
        """Resolve host and port into list of address info entries.

        Translate the host/port argument into a sequence of 5-tuples that contain
        all the necessary arguments for creating a socket connected to that service.
        host is a domain name, a string representation of an IPv4/v6 address or
        None. port is a string service name such as 'http', a numeric port number or
        None. By passing None as the value of host and port, you can pass NULL to
        the underlying C API.

        The family, type and proto arguments can be optionally specified in order to
        narrow the list of addresses returned. Passing zero as a value for each of
        these arguments selects the full range of results.
        """
        # We override this function since we want to translate the numeric family
        # and socket type values to enum constants.
        addrlist = []
>       for res in _socket.getaddrinfo(host, port, family, type, proto, flags):
E       socket.gaierror: [Errno -3] Temporary failure in name resolution

/usr/local/lib/python3.9/socket.py:954: gaierror

During handling of the above exception, another exception occurred:

    @pytest.fixture
    def neo4j():
        """waiting for neo4j to be ready"""
        uri = os.environ.get("NEO4J_URI", "bolt://localhost:7687")
        driver = GraphDatabase.driver(uri)
        session = driver.session()
        connected = False
        while not connected:
            try:
>               session.run("MATCH () return 1 limit 1")

/parser/tests/conftest.py:18:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
/usr/local/lib/python3.9/site-packages/neo4j/_sync/work/session.py:280: in run
    self._connect(self._config.default_access_mode)
/usr/local/lib/python3.9/site-packages/neo4j/_sync/work/session.py:120: in _connect
    super()._connect(access_mode, **access_kwargs)
/usr/local/lib/python3.9/site-packages/neo4j/_sync/work/workspace.py:168: in _connect
    self._connection = self._pool.acquire(**acquire_kwargs_)
/usr/local/lib/python3.9/site-packages/neo4j/_sync/io/_pool.py:411: in acquire
    return self._acquire(
/usr/local/lib/python3.9/site-packages/neo4j/_sync/io/_pool.py:221: in _acquire
    return connection_creator()
/usr/local/lib/python3.9/site-packages/neo4j/_sync/io/_pool.py:138: in connection_creator
    connection = self.opener(
/usr/local/lib/python3.9/site-packages/neo4j/_sync/io/_pool.py:386: in opener
    return Bolt.open(
/usr/local/lib/python3.9/site-packages/neo4j/_sync/io/_bolt.py:322: in open
    BoltSocket.connect(
/usr/local/lib/python3.9/site-packages/neo4j/_async_compat/network/_bolt_socket.py:649: in connect
    for resolved_address in resolved_addresses:
/usr/local/lib/python3.9/site-packages/neo4j/_async_compat/network/_util.py:157: in resolve_address
    for resolved_address in NetworkUtil._dns_resolver(
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

address = IPv4Address(('neo4j', 7687)), family = 0

    @staticmethod
    def _dns_resolver(address, family=0):
        """ Regular DNS resolver. Takes an address object and optional
        address family for filtering.

        :param address:
        :param family:
        :returns:
        """
        try:
            info = NetworkUtil.get_address_info(
                address.host, address.port, family=family,
                type=socket.SOCK_STREAM
            )
        except OSError:
>           raise ValueError("Cannot resolve address {}".format(address))
E           ValueError: Cannot resolve address neo4j:7687

/usr/local/lib/python3.9/site-packages/neo4j/_async_compat/network/_util.py:119: ValueError
_________________________________________________________ ERROR at setup of test_fileiter _________________________________________________________

address = IPv4Address(('neo4j', 7687)), family = 0

    @staticmethod
    def _dns_resolver(address, family=0):
        """ Regular DNS resolver. Takes an address object and optional
        address family for filtering.

        :param address:
        :param family:
        :returns:
        """
        try:
>           info = NetworkUtil.get_address_info(
                address.host, address.port, family=family,
                type=socket.SOCK_STREAM
            )

/usr/local/lib/python3.9/site-packages/neo4j/_async_compat/network/_util.py:114:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

host = 'neo4j', port = 7687

    @staticmethod
    def get_address_info(host, port, *, family=0, type=0, proto=0, flags=0):
>       return socket.getaddrinfo(host, port, family, type, proto, flags)

/usr/local/lib/python3.9/site-packages/neo4j/_async_compat/network/_util.py:102:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

host = 'neo4j', port = 7687, family = 0, type = <SocketKind.SOCK_STREAM: 1>, proto = 0, flags = 0

    def getaddrinfo(host, port, family=0, type=0, proto=0, flags=0):
        """Resolve host and port into list of address info entries.

        Translate the host/port argument into a sequence of 5-tuples that contain
        all the necessary arguments for creating a socket connected to that service.
        host is a domain name, a string representation of an IPv4/v6 address or
        None. port is a string service name such as 'http', a numeric port number or
        None. By passing None as the value of host and port, you can pass NULL to
        the underlying C API.

        The family, type and proto arguments can be optionally specified in order to
        narrow the list of addresses returned. Passing zero as a value for each of
        these arguments selects the full range of results.
        """
        # We override this function since we want to translate the numeric family
        # and socket type values to enum constants.
        addrlist = []
>       for res in _socket.getaddrinfo(host, port, family, type, proto, flags):
E       socket.gaierror: [Errno -3] Temporary failure in name resolution

/usr/local/lib/python3.9/socket.py:954: gaierror

During handling of the above exception, another exception occurred:

    @pytest.fixture
    def neo4j():
        """waiting for neo4j to be ready"""
        uri = os.environ.get("NEO4J_URI", "bolt://localhost:7687")
        driver = GraphDatabase.driver(uri)
        session = driver.session()
        connected = False
        while not connected:
            try:
>               session.run("MATCH () return 1 limit 1")

/parser/tests/conftest.py:18:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
/usr/local/lib/python3.9/site-packages/neo4j/_sync/work/session.py:280: in run
    self._connect(self._config.default_access_mode)
/usr/local/lib/python3.9/site-packages/neo4j/_sync/work/session.py:120: in _connect
    super()._connect(access_mode, **access_kwargs)
/usr/local/lib/python3.9/site-packages/neo4j/_sync/work/workspace.py:168: in _connect
    self._connection = self._pool.acquire(**acquire_kwargs_)
/usr/local/lib/python3.9/site-packages/neo4j/_sync/io/_pool.py:411: in acquire
    return self._acquire(
/usr/local/lib/python3.9/site-packages/neo4j/_sync/io/_pool.py:221: in _acquire
    return connection_creator()
/usr/local/lib/python3.9/site-packages/neo4j/_sync/io/_pool.py:138: in connection_creator
    connection = self.opener(
/usr/local/lib/python3.9/site-packages/neo4j/_sync/io/_pool.py:386: in opener
    return Bolt.open(
/usr/local/lib/python3.9/site-packages/neo4j/_sync/io/_bolt.py:322: in open
    BoltSocket.connect(
/usr/local/lib/python3.9/site-packages/neo4j/_async_compat/network/_bolt_socket.py:649: in connect
    for resolved_address in resolved_addresses:
/usr/local/lib/python3.9/site-packages/neo4j/_async_compat/network/_util.py:157: in resolve_address
    for resolved_address in NetworkUtil._dns_resolver(
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

address = IPv4Address(('neo4j', 7687)), family = 0

    @staticmethod
    def _dns_resolver(address, family=0):
        """ Regular DNS resolver. Takes an address object and optional
        address family for filtering.

        :param address:
        :param family:
        :returns:
        """
        try:
            info = NetworkUtil.get_address_info(
                address.host, address.port, family=family,
                type=socket.SOCK_STREAM
            )
        except OSError:
>           raise ValueError("Cannot resolve address {}".format(address))
E           ValueError: Cannot resolve address neo4j:7687

/usr/local/lib/python3.9/site-packages/neo4j/_async_compat/network/_util.py:119: ValueError
================================================================ warnings summary =================================================================
../usr/local/lib/python3.9/site-packages/_pytest/cacheprovider.py:433
  /usr/local/lib/python3.9/site-packages/_pytest/cacheprovider.py:433: PytestCacheWarning: could not create cache path /.pytest_cache/v/cache/nodeids
    config.cache.set("cache/nodeids", sorted(self.cached_nodeids))

../usr/local/lib/python3.9/site-packages/_pytest/cacheprovider.py:387
  /usr/local/lib/python3.9/site-packages/_pytest/cacheprovider.py:387: PytestCacheWarning: could not create cache path /.pytest_cache/v/cache/lastfailed
    config.cache.set("cache/lastfailed", self.lastfailed)

../usr/local/lib/python3.9/site-packages/_pytest/stepwise.py:52
  /usr/local/lib/python3.9/site-packages/_pytest/stepwise.py:52: PytestCacheWarning: could not create cache path /.pytest_cache/v/cache/stepwise   
    session.config.cache.set(STEPWISE_CACHE_DIR, [])

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
============================================================= short test summary info =============================================================
ERROR ../parser/tests/integration/test_parse_unparse_integration.py::test_round_trip - ValueError: Cannot resolve address neo4j:7687
ERROR ../parser/tests/integration/test_parse_unparse_integration.py::test_two_branch_round_trip - ValueError: Cannot resolve address neo4j:7687    
ERROR ../parser/tests/integration/test_parser_integration.py::test_calling - ValueError: Cannot resolve address neo4j:7687
ERROR ../parser/tests/integration/test_parser_integration.py::test_error_log - ValueError: Cannot resolve address neo4j:7687
ERROR ../parser/tests/unit/test_parser_unit.py::test_normalized_filename - ValueError: Cannot resolve address neo4j:7687
ERROR ../parser/tests/unit/test_parser_unit.py::test_fileiter - ValueError: Cannot resolve address neo4j:7687
=============================================== 1 passed, 3 warnings, 6 errors in 103.71s (0:01:43) ===============================================
make: *** [backend_tests] Error 1
```

The logs are long because it is not able to connect to the server.

---

Sharing the logs to make sure the new syntax does not obstruct the existing flow.

---

The errors are because your Neo4J container is not up. So it's ok.